src/modules/ModuleRunner.ts

```diff
@@ -47,17 +137,22 @@ export class ModuleRunner {
     }
 
     /**
-     * Resets the runner, clearing all known modules.
+     * Exposes all extensions which may be overridden/provided by modules
+     *
+     * @returns En extensionsmanager which exposes the extensions
+     */
+    public get extensions(): ExtensionsManager {
+        return this._extensions;
+    }
+
+    /**
+     * Resets the runner, clearing all known modules, and all extensions
      *
      * Intended for test usage only.
      */
     public reset(): void {
         this.modules = [];
-
-        this.extensions = {
-            cryptoSetup: new DefaultCryptoSetupExtensions(),
-            experimental: new DefaultExperimentalExtensions(),
-        };
+        this._extensions.reset();
```

Comment:
Instead of exposing `ExtensionsManager.reset` "for test usage only", could we instead:

```suggestion
        this._extensions = new ExtensionsManager();
```

?